src/lib/parser.js

```diff
@@ -128,9 +237,12 @@ export default function parse(table, config) {
     const resultsMap = toResultsWithRegExp(settings.results);
     const resultsMapCount = resultsMap.length;
     const columnsWithResultsFilter = getFilterForColumnsWithResults(rows, settings, resultsMap);
-    const results = {};
+    const columnsForNameFilter = settings.displayOpponentNameHint ? getFilterForColumnsWithName(rows, settings) : (cell, index) => false;
```

Comment:
I'd suggest moving condition with `displayOpponentNameHint` into `getFilterForColumnsWithName` to simplify the flow.